tests/control/PlanarArmControllerTest.cpp

```diff
@@ -1,13 +1,60 @@
 #include "../../src/control/PlanarArmController.h"
 
+#include "../../src/Constants.h"
+
+#include <iostream>
+
 #include <catch2/catch.hpp>
 
 using namespace Catch::literals;
 using namespace control;
 using namespace std::chrono_literals;
 
+constexpr double PI = M_PI;
+
+std::string toString(const Eigen::Vector2d& pose) {
+	std::stringstream ss;
+	ss << "(" << pose(0) << ", " << pose(1) << ")";
+	return ss.str();
+}
+
+void assertApprox(const Eigen::Vector2d& p1, const Eigen::Vector2d& p2, double dist = 1e-5,
+				  double angle = 1e-5) {
+	Eigen::Vector2d diff = p1 - p2;
+	bool distEqual = diff.norm() <= dist;
+	double thetaDiff = std::fmod(abs(diff(1)), 2 * PI);
+	// change domain from [0, 2pi) to (-pi, pi]
+	if (thetaDiff > PI) {
+		thetaDiff -= 2 * PI;
+	}
+	bool angleEqual = abs(thetaDiff) <= angle;
```

Comment:
I like that you're thinking about it, but I don't think checking the angle is necessary. At best it's redundant, and at worst it could cause false failures, since near the zero vector angle changes by quite a lot even with small absolute changes.

I think just asserting the norm is less than dist is good enough.